### DISCUSSION TITLE
[Feature] Remove Image from Album While Viewing Image in Album Section

### DISCUSSION THREAD

**Discussion by rjgrandy** [immich-app/immich#13732] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

In the web app, when I am viewing an album and I click on an image. There doesn't seem to be any option to remove the image from the album. I only see an option to delete the image. I have to go back the album view select the image and then remove. When there are duplicate images, this can be tough / and tedious

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by Acidiphilium** [comment]
I must agree about that. Much needed option!

**Comment by bo0tzz** [comment]
#8390

**Comment by Millio345** [comment]
I feel like this request is slightly different from #8390. That one is about removing a photo from an album when reaching it via timeline/search, whereas this is about when you're _already viewing the image from within an album_, i.e., at a URL like /albums/ALBUM-ID/photos/PHOTO-ID.

Since the context is clearly the album, it should be feasible to show a “Remove from album” action directly in the viewer — ideally also with a shortcut. This would make album management much smoother.

**Comment by Sjoerd82** [comment]
I agree that this is not the same as #8390. Flipping through the _full-screen images_ and hitting a _shortcut key_ to remove the current image from the album would be my preferred workflow. I have huge albums (auto-populated by tag using SQL -- also, would be nice if you could create an album from one or more tags), this for me is by far the fastest way to slim down the album.

Currently my workaround is the following. I post this here for people that need this as well (and aren't scared of SQL). Always be careful when manually making changes to the database!

1. Make sure your trash is empty!
2. Flip through the album hitting the "DELETE" key to move images you want out of the album, to trash.
3. Execute the SQL below. It unlinks the images (asset) found in trash from any album it is in.
4. In the front-end, go to trash, and click "Restore All".

```SQL
DELETE FROM albums_assets_assets WHERE "assetsId" IN (
     SELECT aaa."assetsId"
       FROM albums_assets_assets aaa
 INNER JOIN assets ass
         ON ass.id = aaa."assetsId"
      WHERE ass.status = 'trashed'
);
```
